app/Models/Foundation/Summit/Events/SummitEvent.php

```diff
@@ -1739,6 +1762,120 @@ public function promote2Presentation(PresentationType $type):void{
         } catch (\Exception $ex) {
 
         }
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getOverflowStreamingUrl(): ?string
+    {
+        return $this->overflow_streaming_url;
+    }
+
+    /**
+     * @return bool
+     */
+    public function gatOverflowStreamIsSecure(): bool
+    {
+        return $this->overflow_stream_is_secure;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getOverflowStreamKey(): ?string
+    {
+        return $this->overflow_stream_key;
+    }
+
+    /**
+     * @param string $overflow_stream_key
+     * @return void
+     */
+    public function setOverflowStreamKey(string $overflow_stream_key): void
+    {
+        $this->overflow_stream_key = $overflow_stream_key;
+    }
+
+    /**
+     * @param string $overflow_streaming_url
+     * @param bool $overflow_stream_is_secure
+     * @return void
+     */
+    public function setOverflow(string $overflow_streaming_url,
+                                bool $overflow_stream_is_secure): void
+    {
+        $this->overflow_streaming_url = $overflow_streaming_url;
+        $this->overflow_stream_is_secure = $overflow_stream_is_secure;
+        $this->occupancy = self::OccupancyOverflow;
+    }
+
+    public function clearOverflow(): void
+    {
+        $this->overflow_streaming_url = null;
+        $this->overflow_stream_is_secure = false;
+        $this->overflow_stream_key = null;
+        $this->occupancy = self::OccupancyEmpty;
+    }
+
+    /**
+     * @return string
+     */
+    public function getOverflowUrl(): string
+    {
+        return sprintf("%s%s?%s=%s",
+            $this->summit->getMarketingSiteUrl(),
+            config("overflow.path", "/a/overflow-player"),
+            config("overflow.query_string_key","k"),
+            $this->overflow_stream_key);
+    }
+
+    /**
+     * @return string
+     * @throws RandomException
+     */
+    public function generateOverflowKey(): string
+    {
+         $salt = random_bytes(16);
+         return hash('sha256', $this->getId() . $salt . time());
+    }
+
+    /**
+     * @return array
+     */
+    public function getOverflowTokens(): array
```

Comment:
refactoring this to have an unique method
for overflow and for regular stream
please do check https://github.com/OpenStackweb/summit-api/blob/main/app/ModelSerializers/Summit/SummitEventSecureStreamSerializer.php
and unify the methods ( knowledge should be on one single place )